src/nvim/edit.c

```diff
@@ -3947,16 +3950,20 @@ static int ins_compl_get_exp(pos_T *ini)
 /* Delete the old text being completed. */
 static void ins_compl_delete(void)
 {
-  int i;
+  int col;
 
-  /*
-   * In insert mode: Delete the typed part.
-   * In replace mode: Put the old characters back, if any.
-   */
-  i = compl_col + (compl_cont_status & CONT_ADDING ? compl_length : 0);
-  backspace_until_column(i);
-  // TODO: is this sufficient for redrawing?  Redrawing everything causes
-  // flicker, thus we can't do that.
+  // In insert mode: Delete the typed part.
+  // In replace mode: Put the old characters back, if any.
+  col = compl_col + (compl_cont_status & CONT_ADDING ? compl_length : 0);
+  if ((int)curwin->w_cursor.col > col) {
+    if (stop_arrow() == FAIL) {
+      return;
+    }
+    backspace_until_column(col);
+  }
+
+  // TODO(unknown): is this sufficient for redrawing?  Redrawing everything
```

Comment:
The linter complained that there wasn't an assigned user for the `TODO`.  Is assigning it to `unknown` alright?